components/AuthChecker/AuthChecker.tsx

```diff
@@ -0,0 +1,18 @@
+import { useSession } from "@/store/useSession";
+import { useRouter } from "next/router";
+
+type Children = {
+  children: React.ReactNode;
+};
+const AuthChecker = ({ children }: Children) => {
+  const router = useRouter();
+  const { user } = useSession((state) => state);
+  const isLoggedIn = user !== null;
+
+  if (isLoggedIn) {
+    void router.push("/");
+  }
+
+  return <>{!isLoggedIn ? children : null}</>;
```

Comment:
Or better still, instead you could return null from the if (isLoggedIn) check below the router.push call, and then simply return children from the main component body. 

This could provide a clearer flow of logic